server/src/main/java/org/opensearch/monitor/fs/FsHealthService.java

```diff
@@ -134,6 +148,9 @@ public StatusInfo getHealth() {
             statusInfo = new StatusInfo(HEALTHY, "health check disabled");
         } else if (brokenLock) {
             statusInfo = new StatusInfo(UNHEALTHY, "health check failed due to broken node lock");
+        } else if (checkInProgress.get() && currentTimeMillisSupplier.getAsLong() -
```

Comment:
:+1: this works, to be sure - it only flips when `getHealth()` is triggered while `run()` is being executed. Otherwise the FS is assumed to be healthy.

Reply:
Yes that the case @reta

Reply:
I guess I concur that a better way is to mark the node unhealthy until it passes the timeout check to avoid flip flops